server/apps/io/update_ingest_tests.py

```diff
@@ -350,3 +351,93 @@ def test_files_dont_duplicate_ingest(self):
         # 12 files in grid fs
         current_files = self.app.media.fs('upload').find()
         self.assertEqual(12, current_files.count())
+
+    def test_anpa_category_to_subject_derived_ingest(self):
+        vocab = [{'_id': 'categories', 'items': [{'is_active': True, 'name': 'Domestic Sport', 'qcode': 's',
+                                                  "subject": "15000000"}]}]
+        self.app.data.insert('vocabularies', vocab)
+
+        provider_name = 'DPA'
+        guid = 'IPTC7901_odd_charset.txt'
+        provider = get_resource_service('ingest_providers').find_one(name=provider_name, req=None)
+        provider_service = self.provider_services[provider.get('type')]
+        provider_service.provider = provider
+
+        items = provider_service.parse_file(guid, provider)
+        service = get_resource_service('ingest')
+        service.post(items)
+
+        # ingest the items and check the subject code has been derived
+        self.ingest_items(items, provider)
+        self.assertEqual(items[0]['subject'][0]['qcode'], '15000000')
+
+    def test_anpa_category_to_subject_derived_ingest_ignores_inactive_categories(self):
+        vocab = [{'_id': 'categories', 'items': [{'is_active': False, 'name': 'Domestic Sport', 'qcode': 's',
+                                                  "subject": "15000000"}]}]
+        self.app.data.insert('vocabularies', vocab)
+
+        provider_name = 'DPA'
+        guid = 'IPTC7901_odd_charset.txt'
+        provider = get_resource_service('ingest_providers').find_one(name=provider_name, req=None)
+        provider_service = self.provider_services[provider.get('type')]
+        provider_service.provider = provider
+
+        items = provider_service.parse_file(guid, provider)
+        service = get_resource_service('ingest')
+        service.post(items)
+
+        # ingest the items and check the subject code has been derived
+        self.ingest_items(items, provider)
+        self.assertTrue('subject' not in items[0])
```

Comment:
**Tip:** There also exists the `self.assertNotIn()` method (but it's very minor, no need for a change just because of this).